Dockerfile

```diff
@@ -0,0 +1,7 @@
+FROM python:3.12.3
```

Comment:
should this hardcode the patch version of use the latest `3.12` instead ?

Reply:
Not really sure. @acoulton what do you think? 🤔

Reply:
Probably better to use the latest 3.12, that's what our readtheddocs.yaml specifies

Reply:
Updated